static/js/lib/ckeditor/plugins/ResourceLinkMarkdownSyntax.test.ts

```diff
@@ -75,7 +84,9 @@ describe("ResourceLink plugin", () => {
       .processor as unknown) as MarkdownDataProcessor
     expect(md2html('{{< resource_link uuid123 "bad \\" >}}')).toBe(
       // This is wrong. Should not end in &lt;/a&gt;
-      '<p><a class="resource-link" data-uuid="uuid123">bad &lt;/a&gt;</p>'
+      `<p><a class="resource-link" data-uuid="${encode(
```

Comment:
Using `encodeURIComponent(JSON.stringify(args))` for multiple args seems simple, but it means that `'uuid1'` shows up as `'%5B%22uuid1%22%5D'` instead of just `'uuid1'. We could probably avoid this if we really want, but handling 1 args the same as 2 args seems clean, if aesthetically unappealing in tests.

Reply:
I'm a little confused by this, what you're saying is that because it's encoding both `class` and `data-uuid` that it will include the quotations for both as part of the string.?

Reply:
_I'm going to write this a slightly longer reply because I also want to share it with Alice, whom I have not talked to about this change yet but should probably make aware_

I'm not sure what `class` you refer to, but btw: `data-uuid` is an outdated name. Previously it was a uuid. Now it's a string that encodes a uuid and possibly more information (like the anchor). Changing the name of that attribute would break our CKEditor ResourceLink plugin (which is in a different repo, https://github.com/mitodl/ckeditor5). We probably should change the attribute name from `data-uuid` to something else, like `data-link` (which ironically might have been the original name in the Link plugin, I'm not sure). 

If that sounds weird...

My understanding from talking with @alicewriteswrongs is: In CKEditor, links are modeled as an attribute on text and text in CKEditor can only have one attribute. For our resource link stuff, the value of that attribute is determined by the `data-uuid` attribute on the anchor tag.

```
# to/from CKEditor:
Markdown --> HTML --> CKEditor internal representation
         <--      <--
1. convert shortcode to text + metadata (metadata = uuid + anchor)
2. store metadata on anchor data-uuid attribute
3. CKEditor uses the data-uuid attribute as the single attribute for the
   link text in its internal representation
```
The important thing is: In CKEditor's model of a link, it can store the text and one other value. (No more than one other value). For normal links, that "one value" is the url. For our resource_links, that one value was previously the uuid.

But now we need to store two pieces of information in that one value: the uuid and the fragment id.

The way ResourceLink plugin is set up, the "one value" is the same as "data-uuid" attribute on the `<a></a>` tag. 

In Alice's PR (#1037) the two pieces of information were encoded as: `uuid#fragment`, i.e.,  `"piece1#piece2" with `'#'` as the separator. That was really simple but (1) makes including. the '#' piece in the shortcode value slightly harder and also precludes us from using this for search params `?foo=1&bar=2` if we ever needed to. 

So in this PR, I switched the encoding to just use JSON.stringify/parse to encode/decode the multiple pieces (uuid+anchor) to/from a single value. Except the output of JSON.stringify be an HTML attribute value since it has quotations, so I had to use `encodeURIComponent/decode...`. 

This works well for passing 2 (or more) args as a single value to/from CKEditor.

All I meant in my original comment was: **It also works for passing 1 argument, but it's kinda ugly** because the single argument also has the quotations and square brackets from JSON.stringify:
```
const uuid = 'uuid1'
const asJson = JSON.stringify(['uuid1']) // stringify an array for consistency with the two-argument case
// out: the string '["uuid1"]'
const encoded = encodeURIComponent(asJson)
// out: the string '%5B%22uuid1%22%5D'
```

Reply:
@ChristopherChudzicki thanks for the explanation, I now have a greater understanding of how this works and what you meant by a single argument being "ugly."  I don't think it matters as long as the data comes out as expected on the other side.